sql/src/main/java/org/opensearch/sql/sql/parser/AstBuilder.java

```diff
@@ -199,6 +202,11 @@ private UnresolvedExpression visitSelectItem(SelectElementContext ctx) {
     UnresolvedExpression expr = visitAstExpression(ctx.expression());
 
     if (expr instanceof Alias) {
+      List funcArgsParts = ((QualifiedName) ((Function) ((Alias) expr)
+          .getDelegated()).getFuncArgs().get(0)).getParts();
+      if (funcArgsParts.get(funcArgsParts.size() - 1).equals("*")) {
+        return new NestedAllFields(((Alias) expr).getName());
+      }
```

Comment:
Maybe you need to check first whether `delegated` is a function and validate function name.

Reply:
Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/256/commits/14f410c441df9436a0d4605c063923224bb32be8